### DISCUSSION TITLE
Is it possible to set voltage cut-off for drive cycle and how?

### DISCUSSION THREAD

**Discussion by RuiheLi** [pybamm-team/PyBaMM#2965] [Q&A] [answered]
As stated. Thanks a lot! @brosaplanella

**Comment by rtimms** [comment]
See https://github.com/pybamm-team/PyBaMM/issues/1902

**Reply by RuiheLi** [reply]
Thanks a lot! @rtimms

**Reply by RuiheLi** [reply]
Hi @rtimms
sorry to bring it up again.

Just realized that this is not how we did in expeirments. In experiment, we do:
"Run US06_A (A) for 4 times or until 3.7 V", would you have any ideas on how to do that?

**Comment by rtimms** [comment, marked as answer]
FYI the way you pass drive cycles to experiments is about to change https://github.com/pybamm-team/PyBaMM/pull/2960. If you have the drive cycle as an array you could just write something in your script that copies the data for times (and makes the correct adjustment to the time data), but there might be a simpler way.

**Reply by RuiheLi** [reply]
Oh I understand this now. Intially I thought if we say `Run US06_A (A) until 3.7 V` it will just repeat this until 3.7V is reached but after a simple test I found that it only run one times and exit early if 3.7V is hit.
